packages/instantsearch.js/src/lib/__tests__/InstantSearch-test.tsx

```diff
@@ -519,39 +577,31 @@ See https://www.algolia.com/doc/api-reference/widgets/configure/js/`);
         },
       });
 
-      expect(
-        search.middleware.map(({ instance: { $$type, $$internal } }) => ({
-          $$type,
-          $$internal,
-        }))
-      ).toEqual([
+      expect(mapMiddlewares(search.middleware)).toEqual([
         {
           $$type: 'ais.insights',
           $$internal: true,
+          $$automatic: false,
         },
       ]);
     });
 
```

Comment:
You can add a test with `insights: true` + `_automaticInsights: true` to verify the middleware is added only once.

Reply:
Done in [e10f7ea](https://github.com/algolia/instantsearch/pull/5883/commits/e10f7eaab73557f461f36f23ef84b0d9558acbba)